### PR TITLE
refactor: migrate search.js file to typescript

### PR DESCRIPTION
**Pull request by hereje**
#### Summary

 migrate search.js file to typescript

- convert file to typescript
- reorganize into functions
- add types
- add docs

#### Ticket Link

Fixes: https://github.com/mattermost/mattermost/issues/21616

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by agnivade — DISMISSED** [review]
Your branch is messed up and includes other changes in master. Please reset your branch to the latest master and push again.

**Comment by hereje** [comment]
> Your branch is messed up and includes other changes in master. Please reset your branch to the latest master and push again.

Sorry, I didn't notice I messed up with the changes.
Done.

File: e2e-tests/cypress/tests/support/ui/search.ts [hunk 1]
```diff
@@ -19,4 +30,15 @@ Cypress.Commands.add('uiJumpToSearchResult', (postId) => {
 
     // * Verify that the permalinked post is highlighted in the center channel
     cy.get(`#post_${postId}.post--highlight`).should('be.visible');
-});
+}
+
+// eslint-disable-next-line @typescript-eslint/no-use-before-define
+Cypress.Commands.add('uiJumpToSearchResult', uiJumpToSearchResult);
```
Comment (saturninoabril) [inline e2e-tests/cypress/tests/support/ui/search.ts:36]: Let's remove this disable/flag and the other instance in this file by updating the global declaration below with:
```javascript
declare global {
    // eslint-disable-next-line @typescript-eslint/no-namespace
    namespace Cypress {
        interface Chainable {
            uiSearchPosts: typeof uiSearchPosts;
            uiJumpToSearchResult: typeof uiJumpToSearchResult;
        }
    }
}

export {};
```

```suggestion
Cypress.Commands.add('uiJumpToSearchResult', uiJumpToSearchResult);
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @hereje! LGTM, except for couple of request.

File: e2e-tests/cypress/tests/support/ui/search.ts (hunk 1 above)
Comment (hereje) [reply e2e-tests/cypress/tests/support/ui/search.ts]: Thanks for your suggestions.

Done!

**Review by saturninoabril — APPROVED** [review]
Thanks @hereje, LGTM.

**Review by yasserfaraazkhan — APPROVED** [review]

**Comment by saturninoabril** [comment]
@hereje I'm sorry this was overlooked for so long. It appears that the changes have already been implemented/migrated at https://github.com/mattermost/mattermost/pull/28483. I'm closing this issue, but please feel free to reopen it if I've made a mistake.

**Pull request closed without merging**